setuptools/tests/fixtures.py

```diff
@@ -72,3 +56,64 @@ def sample_project(tmp_path):
     except Exception:
         pytest.skip("Unable to clone sampleproject")
     return tmp_path / 'sampleproject'
+
+
+# sdist and wheel artifacts should be stable across a round of tests
+# so we can build them once per session and use the files as "readonly"
+
+
+@pytest.fixture(scope="session")
+def setuptools_sdist(tmp_path_factory, request):
+    with contexts.session_locked_tmp_dir(tmp_path_factory, "sdist_build") as tmp:
+        dist = next(tmp.glob("*.tar.gz"), None)
+        if dist:
+            return dist
+
+        subprocess.check_call([
+            sys.executable, "-m", "build", "--sdist",
+            "--outdir", str(tmp), str(request.config.rootdir)
+        ])
+        return next(tmp.glob("*.tar.gz"))
+
+
+@pytest.fixture(scope="session")
+def setuptools_wheel(tmp_path_factory, request):
+    with contexts.session_locked_tmp_dir(tmp_path_factory, "wheel_build") as tmp:
```

Comment:
I don't think this operation has the desired effect. Since it's a session fixture, I'd expect it to only be run once per session, even if tests are run in parallel, so there shouldn't be any race conditions on generating the artifact even if multiple tests solicit it. Maybe I gave a bad hint when I was pondering the need for a lockfile.